lib/purescript-cst/src/Language/PureScript/AST/Exported.hs

```diff
@@ -30,15 +32,27 @@ import Language.PureScript.Names
 -- list, unless there is no export list, in which case they appear in the same
 -- order as they do in the source file.
 --
+-- Kind signatures declarations are also exported if their associated
+-- declaration is exported.
 exportedDeclarations :: Module -> [Declaration]
 exportedDeclarations (Module _ _ mn decls exps) = go decls
   where
   go = flattenDecls
-        >>> filter (isExported exps)
+        >>> reverse . snd . foldl' (f exps) (Nothing, [])
         >>> map (filterDataConstructors exps)
         >>> filterInstances mn exps
         >>> maybe id reorder exps
 
+  f :: Maybe [DeclarationRef] -> (Maybe Declaration, [Declaration]) -> Declaration -> (Maybe Declaration, [Declaration])
```

Comment:
I would prefer that we update `isExported` to handle kind declarations, rather than duplicating a slightly different version of it inline here. In general, I've found that having two slightly different versions of functions is a really good way to introduce bugs, especially as other changes go in over time. I think the best thing to do is to arrange things so that only one version of the function is necessary, but in cases where that's too difficult, then having one function which can work in both cases is still preferable to duplicating the implementation.

Reply:
Looks like `isExported` is only used for docs. 

How about I convert my inline code into a function called `filterDeclarations` and remove `isExported` completely?

Reply:
Oh wait... Now I think I get what you're saying. My original approach is making this needlessly complicated.

Reply:
FWIW, longer term I think it would be good to remove this whole module, because it duplicates the existing logic in the desugaring pipeline for figuring out what is and isn't exported from a module. Ideally I think we would be able to reuse that logic for docs too, perhaps by looking at externs to figure out what is and isn't exported.